addons/test_mass_mailing/tests/test_performance.py

```diff
@@ -92,3 +92,13 @@ def test_send_mailing_w_bl(self):
 
         self.assertEqual(mailing.sent, 50)
         self.assertEqual(mailing.delivered, 50)
+
+        mails = self.env['mail.mail'].sudo().search([('mailing_id', '=', mailing.id)])
+
+        self.assertEqual(len(mails), 12, 'Should not have auto deleted the blacklisted emails')
+
+        blacklisted_emails = self.env['mail.blacklist'].search([]).mapped('email')
+
+        for mail in mails:
+            self.assertEqual(mail.state, 'cancel', 'Should have kept the cancelled emails')
+            self.assertIn(mail.email_to, blacklisted_emails)
```

Comment:
Maybe we can simplify a bit?

```py
cancelled_mail_count = self.env['mail.mail'].sudo().search([('mailing_id', '=', mailing.id), ('state', '=', 'cancel')])
self.assertEqual(cancelled_mail_count, 12, 'Should not have auto deleted the blacklisted emails')
```

Checking the blacklist content again does not seem necessary, that way we gain some queries during testing.